<a href="https://colab.research.google.com/github/nimrafarooq2006-coder/ml-internship-work/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/nimrafarooq2006-coder/ml-internship-work/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*
                   I picked Lane 2: Refresh / Content Opportunity Scoring.

Why: I already ran the starter notebooks on this problem, and I want to
learn decision trees and random forests. I may change my lane before
the end of Week 4.

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*
Question: Which pages should a reviewer look at first because they may
be losing traffic?

One row is: one web page.

Output: a ranked list of pages, with a short reason for each.

Action: a person reviews the top pages first and decides whether to
update them.

Cost of a wrong call: the reviewer wastes time on a page that was
fine, or misses a page that really needed work.

Why data or ML can help: there are too many pages to check by hand,
and a ranking helps use limited review time.

Why this is not just "train a model": the goal is a better decision
about where people spend their time. The model has to beat a simple
rule that a person could write by hand.

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import os, subprocess
import pandas as pd

path = "data/raw/content_refresh_anonymized.csv"
if not os.path.exists(path):
    if not os.path.isdir("ml-internship-work"):
        subprocess.run(["git", "clone", "--depth", "1",
            "https://github.com/nimrafarooq2006-coder/ml-internship-work"], check=True)
    path = "ml-internship-work/" + path

df = pd.read_csv(path)
df["is_declining"] = df["trend_direction"].str.lower().eq("down").astype(int)
stale = df["days_since_last_update"] >= 180
visible = df["impressions_90d"] >= 500

print("Pages in the data:", len(df))
print("Share of pages declining:", round(df["is_declining"].mean(), 3))
print("Declining rate, stale pages (180+ days):", round(df.loc[stale, "is_declining"].mean(), 3))
print("Declining rate, other pages:", round(df.loc[~stale, "is_declining"].mean(), 3))
print("Pages that are stale AND visible:", int((stale & visible).sum()))
print("Number of stale pages:", int(stale.sum()))


Pages in the data: 30000
Share of pages declining: 0.542
Declining rate, stale pages (180+ days): 0.471
Declining rate, other pages: 0.542
Pages that are stale AND visible: 17
Number of stale pages: 174


Numbers from the starter data: 30,000 pages, and 54.2% are marked
declining. Only 174 pages (about 0.6%) are stale (180+ days since
update), and 17 of those are also visible (500+ impressions).
Stale pages were declining a bit less often (47.1%) than the other
pages (54.2%), but the stale group is small, so this difference may
be noise.

What this suggests: the simple rule "old page + lots of views" picks
out very few pages, so a hand rule alone may miss most declining
pages. This makes the lane worth studying, because a learned ranking
might find signals that a simple rule can't. These results are
directional only.

## 4. Careful words: what I can and can't claim

What I can say: in this data, some page patterns were linked to more
pages marked "declining". A ranking may help decide what to review
first. These results are observed and directional, and meant for
decision support.

What I can't say: that updating a page will fix it, or that I proved
how Google works. This is observational data, not a causal test. The
"declining" label is a simple proxy, not a future outcome.

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.